tests/entrypoint/database/test_parse.py

```diff
@@ -1,15 +1,21 @@
 import pathlib
 import pickle
-
-import pytest
+from argparse import ArgumentError
 
 from bluesearch.database.article import Article
 from bluesearch.entrypoint.database.parent import main
 
 
 def test_unknown_parser():
-    with pytest.raises(ValueError, match="Unsupported parser"):
-        main(["parse", "WrongParser", "path_to_input", "path_to_output"])
+    wrong_parser = "WrongParser"
+    try:
+        main(["parse", wrong_parser, "path_to_input", "path_to_output"])
+
+    # argparse exists with error 2, so we need to "unpack" the exception
+    except SystemExit as e:
+        context = e.__context__
+        assert isinstance(context, ArgumentError)
+        assert f"invalid choice: '{wrong_parser}'" in context.args[1]
```

Comment:
Isn't this going to pass if no `SystemExit` is raised?

Maybe we can do something like this:
```suggestion
    wrong_parser = "WrongParser"
    with pytest.raises(SystemExit) as exc_info:
        main(["parse", wrong_parser, "path_to_input", "path_to_output"])

    # argparse exits with error code 2, so we need to "unpack" the exception
    exc = exc_info.value
    context = exc.__context__
    assert exc.code == 2
    assert isinstance(context, ArgumentError)
    assert f"invalid choice: '{wrong_parser}'" in context.message
```

Reply:
Thanks for the suggestion! Fixed in 8270040.

The only thing that I left as it was is the `context.args[1]` instead of `context.message`. The `message` attribute of `BaseException` was deprecated in Python 2.6 and removed in Python 3.  Now, it is true that some exception types like `ArgumentError` have their own` .message`, but `mypy` and PyCharm don't know what's the type of `context` so they complain about it.

Reply:
Interesting, they do? I would have assumed that `mypy` and `PyCharm` would infer from `assert isinstance(context, ArgumentError)` that `context` is an instance of `ArgumentError`...

Reply:
Actually, I just tried, `mypy` seems to be fine with that. PyCharm complains with the message `Unresolved attribute reference 'message' for class 'ArgumentError'`. Looks like a bug to me.

Another option is to write
```python
assert f"invalid choice: '{wrong_type}'" in str(context)
```

Anyway, not so important. We can resolve :)

Reply:
> I would have assumed that `mypy` and PyCharm would infer from `assert isinstance(context, ArgumentError)` that context is an instance of `ArgumentError`

You're right! I tried w/o the `assert isinstance` and then `mypy` complains, but with that `assert` everything is fine.

Just in case, also to make PyCharm happy, I switched to the `str(context)` you suggested in your last message.

Fixed in 880176ccea2516298652915a524a470dde7cb920.